spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/dsl/MongoDbTests.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright 2007-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mongodb.dsl;
+
+import com.mongodb.MongoClient;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.SimpleMongoDbFactory;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.channel.MessageChannels;
+import org.springframework.integration.handler.ReplyRequiredException;
+import org.springframework.integration.mongodb.test.entity.PersonDomain;
+import org.springframework.integration.mongodb.test.entity.QueryMessage;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.springframework.integration.mongodb.test.MongoDbTestUtils.TEST_COLLECTION;
+import static org.springframework.integration.mongodb.test.MongoDbTestUtils.createPerson;
+
+/**
+ * @author Xavier Padró
+ * @since 5.0
+ */
+@RunWith(SpringRunner.class)
+@DirtiesContext
+public class MongoDbTests {
```

Comment:
The problem with all of these real MongoDB tests that the target environment may not have `mongod` process. Therefore we always use `MongoDbAvailableTests` which includes `MongoDbAvailableRule` JUnit `@Rule`.

From there pay attention that we have `createPerson()` methods, so maybe your new `MongoDbTestUtils` and `PersonDomain` are redundant, too? I mean that when we have these new tests based on the `MongoDbAvailableTests`, we don't need any new testing entities.